### PR TITLE
Validate field length on various fields

### PR DESCRIPTION
**Pull request by djohnston32**
I rule and have completed some work on Case Manager that's ready for review!

Adds validation to various fields to protect against huge values being submitted.

This pull request makes the following changes:
* Adds basic max length validations on a number of fields
* Adds length checks on free-input config options
* Adds length checks on special circumstances

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

File: app/models/config.rb [hunk 1]
```diff
@@ -252,4 +268,13 @@ def validate_patient_archive
     def validate_shared_reset
       validate_number && options.last.to_i.between?(SHARED_MIN_DAYS, SHARED_MAX_DAYS)
     end
+
+    def validate_length
+      total_length = 0
+      options.each do |option|
+        total_length += option.length
```
Comment (colinxfleming) [inline app/models/config.rb:275]: I probably gave unclear directions here, so please forgive me. There are legit reasons for configs to have a total length of 1k+ - for example, at least two funds have a very long list of funds they work with that come out to 1500+ characters. So I think we'll need to adjust the strategy here in particular. We could:
* raise the cap to 4k, like notes are at, and keep this strategy as you have it written here (def fixes it, still keeps us flexible)
* check individual elements against the 150 char limit (mostly fixes the root problem, keeps us flexible)
* say to ourselves 'configs are an admin-only endpoint and so they don't really need validation' (a reasonable approach, would leave only this admin-only module vulnerable to this screwing-around-with)

I'll defer to you on the right path here.

**Review by colinxfleming — CHANGES_REQUESTED** [review]
Excellent work here - I think two minor fixes in configs and then we can ship. Did some smoke testing of other fields and the controls worked perfectly.

File: app/models/config.rb (hunk 1 above)
Comment (djohnston32) [reply app/models/config.rb:275]: Makes sense! I opted for the first option, raising the cap to 4k. I don't feel strongly, but overall I feel like it gives sufficient protection while minimizing the likelihood of false positives compared to a per-element limit.

**Review by colinxfleming — APPROVED** [review]
thank you!

**Pull request merged**